wasm/src/utils.rs

```diff
@@ -1,16 +1,115 @@
 use petgraph::visit::{EdgeRef, IntoEdgeReferences, IntoNodeReferences};
 use std::{collections::VecDeque, error::Error, fmt};
 use wasm_bindgen::prelude::*;
+use web_time::Instant;
 
+#[cfg(not(feature = "test"))]
+#[wasm_bindgen(module = "src/logger/index.ts")]
+extern "C" {
+    #[wasm_bindgen(js_name = log)]
+    pub static LOGGER: Logger;
+
+    pub type Logger;
+    #[wasm_bindgen(method)]
+    pub fn debug(this: &Logger, message: &str);
+    #[wasm_bindgen(method)]
+    pub fn info(this: &Logger, message: &str);
+    #[wasm_bindgen(method)]
+    pub fn warn(this: &Logger, message: &str);
+    #[wasm_bindgen(method)]
+    pub fn error(this: &Logger, message: &str);
```

Comment:
I've also added a `feat` level for features whose purpose is to log things. It has the highest log level